crates/polars-core/src/chunked_array/logical/categorical/mod.rs

```diff
@@ -58,6 +58,23 @@ impl CategoricalChunked {
         &mut self.logical
     }
 
+    pub(crate) fn get_flags(&self) -> u8 {
+        self.bit_settings.bits
+    }
+
+    /// Set flags for the Chunked Array
+    pub(crate) fn set_flags(&mut self, flags: u8) -> PolarsResult<()> {
+        BitSettings::from_bits(flags)
+            .ok_or_else(|| {
+                PolarsError::ComputeError(
```

Comment:
For consistency we should use the `polars_err!` macro. 

replace with `polars_err!(ComputeError: "corrupt flags {} for {}", flags, self.dtype())`